### PR TITLE
Enabling TC to install manylinux2010 dependencies + Dockerfile for CentOS 6

### PR DESCRIPTION
**Pull request by shantanuchhabra**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -205,7 +205,7 @@ build_cpp_linux_no_remotefs:
     DOCKER_DRIVER: overlay2
   script:
     - apt-get -y update
-    - apt-get -y install apt-transport-https ca-certificates gnupg software-properties-common wget bzip2 build-essential xxd
+    - apt-get -y install apt-transport-https ca-certificates gnupg software-properties-common wget bzip2 build-essential xxd libbz2-dev
```
Comment (znation) [inline .gitlab-ci.yml]: Why is this change needed? Nothing else we changed should be related to this part of the build.

File: scripts/install_on_docker.sh [hunk 2]
```diff
@@ -0,0 +1,199 @@
+# vim: set ft=dockerfile:
```
Comment (znation) [inline scripts/install_on_docker.sh]: This script seems redundant with the Dockerfile for CentOS 6, and doesn't seem to be used anywhere. Can we remove it?

File: .gitlab-ci.yml (hunk 1 above)
Comment (abhishekpratapa) [reply .gitlab-ci.yml]: My bad, I'll revert and start a new branch

File: scripts/install_on_docker.sh (hunk 2 above)
Comment (shantanuchhabra) [reply scripts/install_on_docker.sh]: Done!
